lib/pages/reportPark.dart

```diff
@@ -0,0 +1,213 @@
+import 'package:capstone_app/repositories/reportRepository.dart';
+import 'package:capstone_app/style/appTheme.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_calendar_carousel/flutter_calendar_carousel.dart';
+import 'package:capstone_app/repositories/accountRepository.dart';
+import 'package:capstone_app/models/park.dart';
+import 'package:capstone_app/services/auth.dart';
+
+
+import 'package:intl/intl.dart';
+
+class ReportParkPage extends StatefulWidget {
+  @override
+  ReportParkPageState createState() => ReportParkPageState();
+}
+
+class ReportParkPageState extends State<ReportParkPage> {
+  GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  ReportRepository _reportRepo = ReportRepository.getInstance();
+
+  AccountRepository _accountRepo = AccountRepository.getInstance();
+
+  Park _park;
+
+  DateTime _selectedDate;
+
+  TimeOfDay _time = TimeOfDay.now();
+  TimeOfDay picked;
+
+  void _reportGarage() {
+//    TimeOfDay newStartTime = _getTimeOfDayForIndex(_selectedTime.truncate());
+    DateTime reportTime = DateTime(_selectedDate.year, _selectedDate.month, _selectedDate.day, _time.hour, _time.minute);
+    if(reportTime.isAfter(_park.start) && reportTime.isBefore(_park.end) ){
+      _accountRepo.reportTicketForPark(reportTime, _park);
+      print("Reported");
+      SnackBar snackBar = SnackBar(
+        content: Text("Reported"),
+      );
+
+      _scaffoldKey.currentState.showSnackBar(snackBar);
+//      Navigator.of(context).pushReplacementNamed('/history');
+    } else{
+      print("Ticket must be reported between parking hours");
+      SnackBar snackBar = SnackBar(
+        content: Text("Ticket must be between: " + new DateFormat.yMMMd().format(_park.start) + " " + new DateFormat.jm().format(_park.start)
+            + "-"+ new DateFormat.yMMMd().format(_park.end) + " " + new DateFormat.jm().format(_park.end)),
+        duration: Duration(seconds: 7),
+      );
+
+      _scaffoldKey.currentState.showSnackBar(snackBar);
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _selectedDate = _reportRepo.intervalStart;

```

Comment:
The default date should probably be the start date of the park passed in

Reply:
Fixed.